scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -55,24 +55,30 @@ object Common {
     unrollTaps(step.getSources.asScala.toSeq)
 
   /**
-   * Get the total size of the input paths, which may contain a glob
+   * Get the total size of the file(s) specified by the Hfs, which may contain a glob
    * pattern in its path, so we must be ready to handle that case.
    */
-  def inputSizes(step: FlowStep[JobConf]): Seq[(Path, Long)] = {
-    val conf = step.getConfig
-
-    FileInputFormat
-      .getInputPaths(conf)
-      .map { path =>
-        val fs = path.getFileSystem(conf)
-        val size = fs.globStatus(path)
-          .map(status => fs.getContentSummary(status.getPath).getLength)
+  def size(tap: Hfs, conf: JobConf): Long = {
```

Comment:
lets document how this works:

1) Ask the tap for its size
2) If the reported size is 0, as the tap for it's path and check the size ourself